lib/services/elasticsearch.js

```diff
@@ -907,6 +909,302 @@ class ElasticSearch extends Service {
+            _id: extracted.documents[i]._id
+          }
+        });
+
+        // _source: true => makes ES return the updated document source in the
+        // response. Required by the real-time notifier component
+        esRequest.body.push({doc: extracted.documents[i]._source, _source: true});
+        toImport.push(extracted.documents[i]);
+      } else {
+        extracted.rejected.push({document: extracted.documents[i], reason: 'a document ID is required'});
+      }
+    } 
+
+    return this._mexecute(esRequest, toImport, extracted.rejected);
+  }
+
+  /**
+   * Replace multiple documents at once. 
+   * Replacements are rejected if targeted documents do not exist,
+   * but documents in the trashcan are silently revived
+   * (like with the normal "replace" method)
+   * 
+   * @param  {Request} request - Kuzzle API request
+   * @return {Promise} 
+   */
+  mreplace(request) {
+    const 
+      index = request.input.resource.index,
+      type = request.input.resource.collection,
+      toImport = [],
+      extracted = extractMDocuments(request, {
+        _kuzzle_info: {
+          active: true, 
+          author: request.context.user._id ? String(request.context.user._id) : null,
+          updater: null, 
+          updatedAt: null, 
+          deletedAt: null,
+          createdAt: Date.now()
+        }
+      });
+
+    return this.client.mget(extracted.mgetRequest)
+      .then(mgetResult => {
+        const esRequest = initESRequest(request, ['consistency', 'refresh', 'timeout', 'fields']);
+        let idx = 0;
+
+        esRequest.body = [];
+
+        let i; // NOSONAR
+        for (i = 0; i < extracted.documents.length; i++) {
+          const document = extracted.documents[i];
+
+          if (typeof document._id === 'string') {
+            // document._id should always be equal to mgetResult.docs[idx]._id
+            // if the document is not found in ES, then we reject it
+            if (mgetResult.docs[idx].found) {
+              esRequest.body.push({index: {_index: index, _type: type, _id: document._id}});
+              esRequest.body.push(document._source);
+              toImport.push(document);
+            } else {
+              extracted.rejected.push({document, reason: 'cannot replace a non-existing document (use mCreateOrReplace if you need to create non-existing documents)'});
+            }
+
+            idx++;
+          } else {
+            extracted.rejected.push({document, reason: 'a document ID is required'});
+          }
+        }
+
+        return this._mexecute(esRequest, toImport, extracted.rejected);
+      });
+  }
+
+  /**
+   * Delete multiple documents with one request
+   * 
+   * @param  {Request} request - Kuzzle API request
+   * @return {Promise}
+   */
+  mdelete(request) {
+    const 
+      esRequest = initESRequest(request, ['consistency', 'refresh', 'timeout', 'fields']),
+      index = request.input.resource.index,
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use destructuring syntax for these assignments from "request.input.resource". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3514)